src/components/Simulation/index.js

```diff
@@ -18,16 +30,51 @@ const styles = makeStyles(theme => ({
   filtersBtn: {
     marginLeft: theme.spacing(1),
   },
+  spinner: {
+    marginTop: theme.spacing(20),
+  },
 }));
 
 const Transition = React.forwardRef(function Transition(props, ref) {
   return <Slide direction="up" ref={ref} {...props} />;
 });
 
+const mapPeriods = invoices => {
+  const all = invoices.map(invoice => ({
+    key: invoice.period,
+    human: humanize(invoice.period),
+  }));
+  return uniqWith(all, (a, b) => a.key === b.key);
+};
+
+const mapPackages = invoices => {
+  const all = invoices.map(invoice => ({
+    key: invoice.code,
+    human: humanize(invoice.code),
+  }));
+
+  return uniqWith(all, (a, b) => a.key === b.key);
+};
+
+const mapOrgunits = invoices => {
+  const all = invoices.map(invoice => ({
+    key: invoice.orgunit_ext_id,
+    human: invoice.orgunit_name,
+  }));
+
+  return uniqWith(all, (a, b) => a.key === b.key);
+};
```

Comment:
This typically the functions you can extract to /utils and add test for

Reply:
It felt very targeted, so don't know if anything else would have a benefit.

Reply:
Not something else, but mainly to test in isolation. if exported as single named exports. Very easy to test via jest. example here https://bitbucket.org/bluesquare_org/dataviz-portal/src/master/src/lib/graphsUtils.test.js